cadc-inventory/src/main/java/org/opencadc/inventory/InventoryUtil.java

```diff
@@ -123,7 +123,35 @@ public static String computeArtifactFilename(URI uri) {
         }
         return ssp;
     }
-
+    
+    /**
+     * Load and instantiate an instance of the specified Java interface. This method uses
+     * the fully-qualified class name as a system property key; the value of the system property
+     * is the fully qualified class name of an implementation of that interface.
+     * 
+     * @param <T>
+     * @param clazz an interface class
+     * @return configured implementation of the interface
+     * @throws IllegalStateException if an instance cannot be created
+     */
+    public static <T> T loadPlugin(Class<T> clazz) throws IllegalStateException {
+        String cnameProp = clazz.getName();
+        String cname = System.getProperty(cnameProp);
+        if (cname == null) {
+            throw new IllegalStateException("CONFIG: " + cnameProp + " not set");
+        }
+        try {
+            Class c = Class.forName(cname);
+            return (T) c.newInstance();
```

Comment:
As of Java 9, this is deprecated in favour of `c.getDeclaredConstructor().newInstance()`.